### DISCUSSION TITLE
Does WiX Toolset v4 or v3 versions support projects of the "CustomAction" type written in C# within the framework of the ".NET Standard 2.0" or ".NET6" frameworks?

### DISCUSSION THREAD

**Discussion by kns-altx-soft** [wixtoolset/issues#7742] [Questions]
I need to translate an application from ".NET Framework 4.8" to ".NET 6". And it is necessary that when installing the application there are no dependencies on the ".NET Framework". 
WiX Tools v3 and v4 allows you to unpack applications written within the framework of the ".NET 6" framework, but if "CustomAction" is used in the WiX installer, then the dependency on the ".NET Framework 4.8" remains.
I tried to create a "CustomAction" project on ".NET Standard 2.0" and build on its basis "CustomAction.CA.dll" using the following command:

WixToolset.Dtf.MakeSfxCA.exe $(DIR_ProjectBuild)\CustomAction.dll $(DIR_WixToolset.Dtf.CustomAction)\$(Platform)\SfxCA.dll     $(DIR_ProjectPublish)\CustomAction.CA.dll $(DIR_ProjectBuild)\CustomAction.config $(DIR_WixToolset.Dtf.WindowsInstaller)\lib\netstandard2.0 \WixToolset.Dtf.WindowsInstaller.dll <Set of libraries used in the CustomAction project >

After packing the "CustomAction" project into "CustomAction.CA.dll" this library is used in the WiX v4 installer and works stably, but only if the «.NET Framework» framework is available.

After a series of experiments, I concluded that the library "SfxCA.dll" dependent on the framework ".NET Framework". And the "CustomAction" assembly is possible only for the ".NET Framework" framework, although the «WixToolset.Dtf.WindowsInstaller.dll» library is built for the ".NET Standard 2.0" and ".NET Framework 2" frameworks.

If there is no support for projects of the type "CustomAction" written in C# within the framework of the frameworks ".NET Standard 2.0" or ".NET6", then how is it currently accepted to transfer the functionality of projects "CustomAction" written within the framework of the frameworks ".NET Framework", under programs that are written within the framework of the frameworks ".NET Standard 2.0" or ".NET 6". And also the answer, is it planned to translate the utility WixToolset.Dtf.MakeSfxCA.exe and libraries SfxCA.dll to the frameworks "NET Standard 2.0" or ".NET 6", so that it would be possible to assemble projects of the type "CustomAction" on these frameworks?

**Comment by chrpai** [comment]
I don't recall any support for .NET (core) being announced.  Do your custom actions have any dependencies that can't run on .NET Framework 4.61+ / .NET Standard 2.0 or is it more of a requirement to get rid of .NET Framework?  You could just rewrite all your CAs in C++.

**Reply by nickhoeferpickpro** [reply]
then how do we do custom actions for .net core?
